### PR TITLE
Refactor common/types into individual packages under pkg/

### PR DESCRIPTION
**Pull request by tgraf**
Creates the following new packages:
 * pkg/endpoint
 * pkg/labels
 * pkg/mac
 * pkg/option
 * pkg/policy

Signed-off-by: Thomas Graf <thomas@cilium.io>

### CODE REVIEW

**Comment by aanm** [comment]
@tgraf tests are failing

File: common/types/container.go [hunk 1]
```diff
@@ -18,11 +18,13 @@ package types
 import (
 	dTypes "github.com/docker/engine-api/types"
 	k8sDockerLbls "k8s.io/client-go/1.5/pkg/kubelet/types"
+
+	"github.com/cilium/cilium/pkg/labels"
```
Comment (aanm) [inline common/types/container.go]: Please put this line on top, dependencies order:
```
- golang

- project

- others
```

File: pkg/labels/filter.go [hunk 2]
```diff
@@ -97,11 +97,11 @@ func ReadLabelPrefixCfgFrom(fileName string) (*LabelPrefixCfg, error) {
 	return &lpc, nil
 }
 
-// FilterLabels returns Labels from the given labels that have the same source and the
+// FilternLabels returns Labels from the given labels that have the same source and the
```
Comment (aanm) [inline pkg/labels/filter.go]: What happen here?

**Comment by aanm** [comment]
I think the package `mac` is too specific. We still have some types about IP's and L3/L4 addresses, should we move those ones to a more abstract package including the contents of `mac`?

File: common/types/container.go (hunk 1 above)
Comment (tgraf) [reply common/types/container.go]: done

File: pkg/labels/filter.go (hunk 2 above)
Comment (tgraf) [reply pkg/labels/filter.go]: probably regex replace oversight

**Comment by tgraf** [comment]
I don't think it makes sense to abstract MACs and IPs into a common type. The suggested `mac` package is a clean and simple extension of net.HardwareAddr to provide Marshal/Unmarshal and a way to convert into `uint64_t`.

I'm fine with moving the IP addressing helpers and abstractions into a package as well but I think it should be separate from the MAC stuff.

**Review by aanm — APPROVED** [review]

**Pull request merged**